sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/PaginationCarbonReader.java

```diff
@@ -174,14 +170,15 @@ private Range getBlockletIndexRange(long fromRowNumber, long toRowNumber) {
         BlockletDetailInfo detailInfo =
             ((CarbonInputSplit) allBlockletSplits.get(i)).getDetailInfo();
         int rowCountInBlocklet = detailInfo.getRowCount();
-        Object[] rowsInBlocklet = new Object[rowCountInBlocklet];
+        // Object[] rowsInBlocklet = new Object[rowCountInBlocklet];
+        List<Object> rowsInBlocklet = new ArrayList<>();
```

Comment:
a. why not use array itself ?
b. Don't keep commented code
c. int count = 0; This is unused now ?

Reply:
also check rowCountInBlocklet

Reply:
Using array is causing the problem in case of update or delete operation.
Removed unused code.